main-command/src/main/scala/sbt/internal/client/NetworkClient.scala

```diff
@@ -328,13 +328,42 @@ class NetworkClient(
             term.isSupershellEnabled
           ).mkString(",")
 
-        val cmd = arguments.sbtLaunchJar match {
+        val (cmd: List[String], envs: Map[String, String]) = arguments.sbtLaunchJar match {
           case Some(lj) =>
-            List("java") ++ arguments.sbtArguments ++
-              List("-jar", lj, DashDashDetachStdio, DashDashServer)
+            def loadConfigFile(file: File): List[String] =
+              IO.readLines(file)
+                .map(_.trim)
+                .filter(_.nonEmpty)
+                .filterNot(_.startsWith("#"))
+
+            val jvmOpts =
+              loadConfigFile(file(".jvmopts"))
+            val sbtOpts =
+              sys.env.get("SBT_OPTS").toSeq.flatMap(_.split(" ").map(_.trim)) ++
```

Comment:
Ii's hard to reproduce shell globbing, so this will incorrectly split `SBT_OPTS=--sbt-dir /dir`, but as the sbt script flags take only path or integer params, they should not pass the shortlist of prefix below.